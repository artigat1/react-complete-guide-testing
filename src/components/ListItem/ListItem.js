import React from 'react';

const TodoListItem = props => {
    const onClickClose = () => {
        const index = parseInt(props.index);
        props.removeItem(index);
    };

    const onClickDone = () => {
        const index = parseInt(props.index);
        props.markTodoDone(index);
    };

    const todoClass = props.item.done ? 'done' : 'undone';
    return (
        <li className="list-group-item ">
            <div className={todoClass}>
                <button type="button" className="close" onClick={onClickClose}>
                    &times;
                </button>
                &nbsp;
                <span
                    className="glyphicon glyphicon-ok icon"
                    aria-hidden="true"
                    onClick={onClickDone}
                >
                    {props.item.value}
                </span>
            </div>
        </li>
    );
};

export default TodoListItem;
