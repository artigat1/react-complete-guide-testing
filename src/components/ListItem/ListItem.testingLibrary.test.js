import React from 'react';
import { render } from '@testing-library/react';

import ListItem from './ListItem';
import renderer from 'react-test-renderer';

describe('<ListItem>', () => {
    let utils;
    let item;
    const idx = 3;
    const mockCloseClickHandler = jest.fn();
    const mockDoneClickHandler = jest.fn();

    beforeEach(() => {
        mockCloseClickHandler.mockClear();
        mockDoneClickHandler.mockClear();

        item = { index: idx, value: 'learn react', done: false };
        utils = render(
            <ListItem
                item={item}
                index={idx}
                key={idx}
                removeItem={mockCloseClickHandler}
                markTodoDone={mockDoneClickHandler}
            />
        );
    });

    it('should render correctly', () => {
        const component = renderer
            .create(<ListItem item={item} index={idx} key={idx} />)
            .toJSON();
        expect(component).toMatchSnapshot();
    });

    it('should render item', () => {
        console.log(utils);
        expect(utils.queryByText(item.value)).toHaveLength(1);
    });
});
