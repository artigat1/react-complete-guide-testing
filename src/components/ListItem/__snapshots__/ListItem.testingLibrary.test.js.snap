// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ListItem> should render correctly 1`] = `
<li
  className="list-group-item "
>
  <div
    className="undone"
  >
    <button
      className="close"
      onClick={[Function]}
      type="button"
    >
      ×
    </button>
     
    <span
      aria-hidden="true"
      className="glyphicon glyphicon-ok icon"
      onClick={[Function]}
    >
      learn react
    </span>
  </div>
</li>
`;
